### DISCUSSION TITLE
Can I filter Area Path before start loading ?

### DISCUSSION THREAD

**Discussion by helenoalves** [nbrown02/FlowViz#33] [Q&A] [answered]
First of all thank you for your amazing work.

On the context that I`m working I have one project and a lot of area paths. I would like to filter the area paths in the initial wizard before loading all data. It will make the tool more usable on my context.
I created my own version of Ageing Chart, and some Control Chart based on Azure DevOps Analitics Views, but I would like to try FlowViz, but I need a pre filter on area path to avoid loading all org data.

Best Regards

**Comment by nbrown02** [comment]
Thanks for the kind words @helenoalves :)

That's a good question - I know it can be done within the OData queries, however I'm not sure how it would be able to be handled from a parameters perspective. For example if users were to leave the parameter for area path as blank, I think the query may fail...but let me do some digging and get back to you 👍

**Reply by nbrown02** [reply, marked as answer]
I just did a quick test using my dummy project:

- Test to see if queries by area path work (where TestArea would be a parameter users would enter)

```
  https://analytics.dev.azure.com/n123btesting/FlowViz/_odata/v3.0-preview/WorkItems? 
$filter=StateCategory ne 'Completed'
            and startswith(Area/AreaPath,'TestArea')
```

This works, so we know we can filter by AreaPath in our queries

- Test to see if users did not enter an Area Path what would happen

```
  https://analytics.dev.azure.com/n123btesting/FlowViz/_odata/v3.0-preview/WorkItems? 
$filter=StateCategory ne 'Completed'
            and startswith(Area/AreaPath,'')
```

This also works - so this looks like it can be done!

**Reply by nbrown02** [reply]
I would need to have Area Path as an optional parameter for users to enter. Let me do some proper tests and then we can get it up and running :)

**Reply by nbrown02** [reply]
So after some further research, doesn't look like this is possible 👎 

It works fine when you have an Area Path
![image](https://user-images.githubusercontent.com/29369962/114441295-1c42cd00-9bc3-11eb-9458-a6883d6fa477.png)

![image](https://user-images.githubusercontent.com/29369962/114441342-2c5aac80-9bc3-11eb-882a-c165eb754b54.png)

However when you leave it blank, it creates issues:

![image](https://user-images.githubusercontent.com/29369962/114441387-3da3b900-9bc3-11eb-80fd-6edcb007a81a.png)

![image](https://user-images.githubusercontent.com/29369962/114441415-45fbf400-9bc3-11eb-93d2-82abaf7ebee3.png)

![image](https://user-images.githubusercontent.com/29369962/114441587-7ba0dd00-9bc3-11eb-9168-045ce0070eb0.png)

Any ideas?

**Reply by helenoalves** [reply]
Hi @nbrown02,

I did this way:
1. I created one new parameter with default value %20:

![Captura de tela 2021-04-12 205003](https://user-images.githubusercontent.com/2830215/114480694-6f277f00-9bd9-11eb-86ba-20e89af56bc9.png)

![Captura de tela 2021-04-12 205039](https://user-images.githubusercontent.com/2830215/114480724-7f3f5e80-9bd9-11eb-8970-62cd2c537734.png)

2. And I changed every odata query putting the filter as the second part:

![Captura de tela 2021-04-12 204911](https://user-images.githubusercontent.com/2830215/114480764-92522e80-9bd9-11eb-8a1a-6cffa06f1ebc.png)

It seems to be ok !
In my context I have more than one AreaPath to include :)
Thank you for your support.

**Reply by nbrown02** [reply]
Amazing! Great stuff 🙂
